src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -31,159 +32,278 @@ public ManagedIdentityAuthRequest(
+            {
+                AuthenticationRequestParameters.RequestContext.ApiEvent.CacheInfo = CacheRefreshReason.NoCachedAccessToken;
             }
 
-            return authResult;
+            logger.Info("[ManagedIdentityRequest] No cached access token found. Getting a token from the managed identity endpoint.");
+
+            return await AcquireFreshTokenAsync(
+                CacheRefreshReason.NoCachedAccessToken,
+                "[ManagedIdentityRequest] Cache miss; acquiring new token.",
+                popRequested,
+                cancellationToken,
+                logger).ConfigureAwait(false);
         }
 
-        private async Task<AuthenticationResult> GetAccessTokenAsync(
+        private async Task<AuthenticationResult> AcquireFreshTokenAsync(
+            CacheRefreshReason cacheReason,
+            string logMessage,
+            bool popRequested,
             CancellationToken cancellationToken,
             ILoggerAdapter logger)
         {
-            AuthenticationResult authResult;
-            MsalAccessTokenCacheItem cachedAccessTokenItem = null;
+            if (!string.IsNullOrEmpty(logMessage))
+            {
+                AuthenticationRequestParameters.RequestContext.ApiEvent.CacheInfo = cacheReason;
+                logger.Info(logMessage);
+            }
 
-            // Requests to a managed identity endpoint must be throttled; 
-            // otherwise, the endpoint will throw a HTTP 429.
+            // Throttle network calls to MI endpoint to avoid HTTP 429s
             logger.Verbose(() => "[ManagedIdentityRequest] Entering managed identity request semaphore.");
             await s_semaphoreSlim.WaitAsync(cancellationToken).ConfigureAwait(false);
             logger.Verbose(() => "[ManagedIdentityRequest] Entered managed identity request semaphore.");
 
             try
             {
-                // While holding the semaphore, decide whether to bypass the cache.
-                // Re-check because another thread may have filled the cache while we waited.
-                // Bypass when:
-                // 1) ForceRefresh is requested
-                // 2) Proactive refresh is in effect
-                // 3) Claims are present (revocation flow)
-                if (_managedIdentityParameters.ForceRefresh ||
-                    AuthenticationRequestParameters.RequestContext.ApiEvent.CacheInfo == CacheRefreshReason.ProactivelyRefreshed ||
-                    !string.IsNullOrEmpty(_managedIdentityParameters.Claims))
+                // If we came here due to a cache miss, re-check after acquiring the semaphore
+                if (cacheReason == CacheRefreshReason.NoCachedAccessToken)
                 {
-                    authResult = await SendTokenRequestForManagedIdentityAsync(logger, cancellationToken).ConfigureAwait(false);
+                    var recheck = await GetCachedAccessTokenAsync().ConfigureAwait(false);
```

Comment:
This will fetch a bearer token even in the POP case. I think you are missing a test like: 

- get a bearer token
- get a pop token (and expect it to get it from STS)